pkg/cache/cache_test.go

```diff
@@ -0,0 +1,211 @@
+package cache_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/cache"
+	"github.com/openshift-psap/special-resource-operator/pkg/clients"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestCache(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cache Suite")
+}
+
+func resetGlobals() {
+	clients.Interface = nil
+
+	cache.Node.Count = 0xDEADBEEF
+	cache.Node.List = &unstructured.UnstructuredList{
+		Object: map[string]interface{}{},
+		Items:  []unstructured.Unstructured{},
+	}
+}
+
+var _ = Describe("Cache", func() {
+	Context("Nodes", func() {
+		BeforeEach(resetGlobals)
+		AfterEach(resetGlobals)
```

Comment:
Why do we need it before and after?

Reply:
- Before: to make sure the global variable is nil
- After: not to contaminate any other test that might be using that global variable without resetting it properly

Do you see things otherwise?

Reply:
But the "other test" will reset it as well because it is in the Before function - unless I missed something

Reply:
Functions in other packages might also use global variables defined in the `cache` package - tests there will not be covered by the `BeforeEach` here. If they don't properly reset `cache` globals variables, we might run into a problem.

Reply:
Then we can have it only in `AfterEach`, unless you expect the tests to run in parallel but then we will have other issues.

Reply:
Fair enough. This is what I was doing in the `testing` version.

Reply:
Would storing the contents at the beginning to set them again at the end of the test make sense? So as to not wipe out something that some other test might do. Leave things as they were found, in other words.

Reply:
I think in this case every `It` sets the globals to test specific cases, so I just left `AfterEach`.